lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1337,6 +1341,23 @@ class HSADevice final : public KalmarDevice
         }
 
         return checksum.str();
+#else
+        // FNV-1a hashing, 64-bit version
+        const uint64_t FNV_prime = 0x100000001b3;
+        const uint64_t FNV_basis = 0xcbf29ce484222325;
+        uint64_t hash = FNV_basis;
+
+        const char *str = static_cast<const char *>(source);
+
+        // 104 is the proper size from Jack's research
```

Comment:
For reference purpose, here are sources where I got this 104 magic number. It's the larger value of these 2 kinds of headers.

- HSA BRIG module header: http://www.hsafoundation.com/html/Content/PRM/Topics/18_BRIG/BrigModuleHeader.htm
- ELF64 header: https://refspecs.linuxfoundation.org/elf/gabi4+/ch4.eheader.html